r/src/arrow_types.h

```diff
@@ -27,10 +27,13 @@
 #include <arrow/ipc/writer.h>
 #include <arrow/type.h>
 
-#define R_ERROR_NOT_OK(s)                  \
-  do {                                     \
-    if (!s.ok()) Rcpp::stop(s.ToString()); \
-  } while (0);
+#define STOP_IF_NOT(TEST, MSG)  \
+  do {                          \
+    if (!TEST) Rcpp::stop(MSG); \
+  } while (0)
+
+#define STOP_IF_NOT_OK(s) STOP_IF_NOT(s.ok(), s.ToString())
+#define STOP_IF_NULL(buf) STOP_IF_NOT(buf, "invalid data")
```

Comment:
At some point we might try to make this into a more informative error message about what's wrong. I can imagine a user getting this error and saying "what the heck"